### DISCUSSION TITLE
Using minio presigned url to upload big files (size around 10 gb)

### DISCUSSION THREAD

**Discussion by prasanjeetchdeep** [minio/minio#12194] [Q&A] [answered]
Hello,

i am currently working on a project where i need to upload a large file to minio. For this i am receiving a presigned url created from another backend service (this service initiates a minio client and generates the preaigned url).
 My objective is uploading the big file(s) to minio bucket . If i try to upload the file by just using the whole file at a time(and not break into chunks), i presume the browser will crash.. so how can i upload large file to minio bucket. 

size of file may vary from 5-10gb

Thanks

**Comment by harshavardhana** [comment, marked as answer]
Presigned PUT doesn't allow more than 5G you are advised to use presigned with multipart uploads https://docs.aws.amazon.com/AmazonS3/latest/userguide/PresignedUrlUploadObject.html - we comply with these docs.

**Comment by prasanjeetchdeep** [comment]
Thank you for the answer.
I have some doubts still
*The help file i am referring
: https://docs.min.io/docs/upload-files-from-browser-using-pre-signed-urls.html
<https://docs.min.io/docs/upload-files-from-browser-using-pre-signed-urls.html>
*
Please note : i am using javascript to upload the files.

*Case 1) * if we have files below 5GB , then can we use this procedure , i
mean directly use fetch to upload files below 5GB
`fetch(url, { method: 'PUT', body: file }).then(() => { // If multiple files
are uploaded, append upload status on the next line. //action on file
upload }).catch((e) => { console.error(e); });
`
i mean logically 5GB is still a big file to upload , will it not crash the
browser ? * Case 2) *If the file is above 5GB : Are you suggesting we
create a presigned URl with multipart uploads as mentioned in the document
you shared ?
how will we be using a generated presigned URL in that case to upload big
files to minio using javascript.
will the files be automatically broken into chunks ? or how will it be
possible without breaking the browser.? Can you please help with any
documentation for this ?
